apps/passport-server/src/routing/routes/telegramRoutes.ts

```diff
@@ -275,4 +280,28 @@ export function initTelegramRoutes(
       res.status(500).send(errorHtmlWithDetails(e as Error));
     }
   });
+
+  app.get("/telegram/user", async (req: Request, res: Response) => {
+    try {
+      const telegramId = checkOptionalQueryParam(req, "telegramId");
+      const semaphoreId = checkOptionalQueryParam(req, "semaphoreId");
+      if (telegramId || semaphoreId) {
+        const db = await getDB();
+        if (telegramId) {
+          const user = await fetchUserByTelegramId(db, telegramId);
+          if (!user) throw new Error(`No user found`);
+          res.json(user);
```

Comment:
not a good idea. there is a public endpoint which returns a list of all semaphore ids. one could download that, then hit this endpoint with the semaphore id to download each user's email address. this would leak user information.

if you want this endpoint to be publicly accessible, you need to carefully redact the user you return. idk if that's even possible.

otherwise, i suggest gating this via some secret 'api key' that can be set via an environment variable.

Reply:
actually looking at this again `fetchUserByTelegramId` et. al. doesn't fetch user, it fetches from the telegram bot conversations table. suggest renaming the function to something more accurate.

Reply:
Didn't know there was that list of public endpoints! I think an api key is good.

Reply:
It's the semaphore group api, which is used for zupoll.

Reply:
🤦